integrations/jira/jira/client.py

```diff
@@ -54,11 +52,103 @@ def __init__(self, jira_url: str, jira_email: str, jira_token: str) -> None:
             self.jira_api_auth = BasicAuth(self.jira_email, self.jira_token)
 
         self.api_url = f"{self.jira_rest_url}/api/3"
+        self.teams_base_url = f"{self.jira_url}/gateway/api/public/teams/v1/org"
         self.webhooks_url = f"{self.jira_rest_url}/webhooks/1.0/webhook"
 
         self.client = http_async_client
         self.client.auth = self.jira_api_auth
         self.client.timeout = Timeout(30)
+        self.semaphore = asyncio.Semaphore(MAX_CONCURRENT_REQUESTS)
+
+    async def _send_api_request(
+        self,
+        method: str,
+        url: str,
+        params: Optional[dict[str, Any]] = None,
+        json: Optional[dict[str, Any]] = None,
+        headers: Optional[dict[str, str]] = None,
+    ) -> Any:
+        async with self.semaphore:
+            response = await self.client.request(
+                method=method, url=url, params=params, json=json, headers=headers
+            )
+            response.raise_for_status()
```

Comment:
This block should be wrapped in a try-catch pair to ensure errors are caught and adequate logs are put on the console.